app/search/search.py

```diff
@@ -20,6 +20,7 @@ def __init__(self, **kwargs):
         self.keywords = kwargs.pop('keywords', [])
         self.origin = kwargs.pop('origin', [])
         self.id = kwargs.pop('id')
+        self.original_dict = kwargs.pop('original_dict', None)
```

Comment:
Please name `original_dict` something like `datasource`, - we're never going to remember what `original_dict` refers to in 2 months.

Reply:
True that, i doubt i would i remember the name.